cocotb/handle.py

```diff
@@ -625,7 +637,19 @@ def setimmediatevalue(self, value):
             self._log.critical("Unsupported type for value assignment: %s (%s)", type(value), repr(value))
             raise TypeError("Unable to set simulator value with type %s" % (type(value)))
 
-        simulator.set_signal_val_binstr(self._handle, value.binstr)
+        simulator.set_signal_val_binstr(self._handle, set_action, value.binstr)
+
+    def check_for_set_action(self, value):
+        if isinstance(value,Freeze):
```

Comment:
nit: space before comma here and in other places in this patch.

```suggestion
        if isinstance(value, Freeze):
```